test/functional/tests/failover_standby/test_failover_advanced.py

```diff
@@ -212,4 +771,212 @@ def test_failover_io_long(pyocf_ctx, cls, cleaning_policy):
+    ):
+        secondary_node.cache.standby_detach()
+
+    with TestRun.use_dut(secondary_node), TestRun.step(
+        "Activate standby cache instance directly on the cache RAID drive"
+    ):
+        secondary_node.cache.standby_activate(secondary_node.raid)
+
+    with TestRun.use_dut(secondary_node), TestRun.step(f"Mount file system"):
+        mountpoint = "/tmp/standby_io_test_mount_point"
+        TestRun.executor.run(f"rm -rf {mountpoint}")
+        create_directory(path=mountpoint)
+        core.mount(mountpoint)
+
+    with TestRun.use_dut(secondary_node), TestRun.step("Calculate checksum of CAS exported object"):
+        checksum2 = File(file_path).md5sum()
+
+    with TestRun.use_dut(primary_node), TestRun.step("Wait for the primary DUT to be back online"):
+        TestRun.executor.wait_for_connection()
+
+    with TestRun.step("Verify that the two checksums are equal"):
+        assert checksum1 == checksum2
+
+
+def check_drbd_installed(duts):
+    for dut in duts:
+        with TestRun.use_dut(dut):
+            if not Drbd.is_installed():
+                TestRun.fail(f"DRBD is not installed on DUT {dut.ip}")
+
+
+def prepare_devices(duts):
+    for dut in duts:
+        with TestRun.use_dut(dut):
+            TestRun.dut.hostname = TestRun.executor.run_expect_success("uname -n").stdout
+
+            raid_members = [TestRun.disks["raid_dev1"], TestRun.disks["raid_dev2"]]
+            for d in raid_members:
+                # 10% extra space for RAID metadata
+                d.create_partitions([raid_size * 1.1, metadata_size])
+
+            raid_config = RaidConfiguration(
+                level=Level.Raid1,
+                metadata=MetadataVariant.Legacy,
+                number_of_devices=2,
+                size=raid_size,
+            )
+            dut.raid = Raid.create(raid_config, [d.partitions[0] for d in raid_members])
+            dut.raid_path = readlink(dut.raid.path)
+
+            dut.cache_md_dev = raid_members[0].partitions[1]
+            dut.core_md_dev = raid_members[1].partitions[1]
+
+            TestRun.disks["core_dev"].create_partitions([core_size])
+            dut.core_dev = TestRun.disks["core_dev"].partitions[0]
+
+
+def get_drbd_configs(primary, secondary):
+    cache_drbd_nodes = [
+        Node(primary.hostname, primary.raid_path, primary.cache_md_dev.path, primary.ip, "7790"),
+        Node(
+            secondary.hostname,
+            cache_exp_obj_path,
+            secondary.cache_md_dev.path,
+            secondary.ip,
+            "7790",
+        ),
+    ]
+    core_drbd_nodes = [
+        Node(dut.hostname, dut.core_dev.path, dut.core_md_dev.path, dut.ip, "7791")
+        for dut in [primary, secondary]
+    ]
+
+    cache_drbd_resource = Resource(name="caches", device="/dev/drbd0", nodes=cache_drbd_nodes)
+
+    core_drbd_resource = Resource(name="cores", device="/dev/drbd100", nodes=core_drbd_nodes)
+
+    return cache_drbd_resource, core_drbd_resource
```

Comment:
general comment: sometimes within TestRun.step there is "):" left in the end of line, and sometimes moved to a new line. It would be great to make it to look alike ( I prefer leaving it at the end of line/ step text) 🌼

Reply:
It is consistent: whenever TestRun.step is multiline, the ending parenthesis is in a newline. I would prefer to keep it this way as this is how the Black tool formats the code It is very convenient to depend on this tool instead of manually adjusting all the steps.

Reply:
after checking the black tool I see, thanks!:)